.github/actions/pr-approvals/action.yml

```diff
@@ -0,0 +1,16 @@
+name: 'PR Approvals'
+description: 'Get the amount of PR approvals'
+runs:
+  using: "composite"
+  steps:
+    - name: Get the amount of PR approvals
+      shell: bash
+      run: |
+        if [ "${{ github.event.review.state }}" == "approved" ]; then
+          pr_approvals=$(gh api "${GITHUB_API_URL}/repos/${GITHUB_REPOSITORY}/pulls/${{ github.event.pull_request.number }}/reviews" | jq '[.[] | select(.state == "APPROVED")] | length')
+          echo "PR approvals count: $pr_approvals"
+          echo "pr_approvals=$pr_approvals" >> $GITHUB_ENV
+        else
+          echo "There are no PR approvals"
+          echo 'pr_approvals=0' >> $GITHUB_ENV
+        fi
```

Comment:
It will be run on every approval, but only the first approval will run xcmetrics. Not ideal, but I tried to limit the runs as much as possible. Let me know wdyt.

Reply:
Do we need to wait for the approvals though? Might slow down the flow a bit. My proposal is: 
- once when the PR is raised 
- if there are major changes in the PR - we should do manual invocation
- we have nightly checks
- we have release checks
Although, since we usually do manual QA after approval, maybe this won't have impact - so both options are ok imo.

Reply:
@testableapple Yeah I agree with @martinmitrevski, probably it is better to just do it when the PR is open 🤔

Reply:
Updated 👍

So now the workflow will be triggered once:
- Nightly
- On any non-draft PR